pkg/clustermanager/cluster_manager.go

```diff
@@ -709,11 +710,13 @@ func (c *ClusterManager) waitForControlPlaneReplicasReady(ctx context.Context, m
 
 func (c *ClusterManager) waitForMachineDeploymentReplicasReady(ctx context.Context, managementCluster *types.Cluster, clusterSpec *cluster.Spec) error {
 	var machineDeploymentReplicasCount int
+	readyReplicas, totalReplicas := 0, 0
 	for _, workerNodeGroupConfiguration := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
 		machineDeploymentReplicasCount += workerNodeGroupConfiguration.Count
 	}
 
 	isMdReady := func() error {
+		readyReplicas, totalReplicas, _ = c.countReplicasReady(ctx)
 		return c.clusterClient.ValidateWorkerNodes(ctx, clusterSpec.Cluster.Name, managementCluster.KubeconfigFile)
```

Comment:
I'm not sure I follow this
Shouldn't `readyReplicas == totalReplicas` be enough here?
Maybe I don't understand what this is doing